packages/wonder-blocks-theming/src/utils/__tests__/merge-theme.test.ts

```diff
@@ -0,0 +1,87 @@
+import tokens from "../../tokens";
+import {mergeTheme} from "../merge-theme";
+
+describe("mergeTheme", () => {
+    it("should return the source theme if the target is empty", () => {
+        // Arrange
+        const themeDefault = {
+            color: {
+                bg: {
+                    default: "#ffffff",
+                    primary: "#0000ff",
+                },
+            },
+            spacing: {
+                small: 4,
+                medium: 8,
+            },
+        };
+        const themeOverride = {};
+
+        // Act
+        const result = mergeTheme(themeDefault, themeOverride);
+
+        // Assert
+        expect(result).toEqual(themeDefault);
+    });
+
+    it("should merge two themes", () => {
+        // Arrange
+        const themeDefault = {
+            color: {
+                blue: "#0000f0",
+                green: "#00ff00",
+            },
+            spacing: {
+                medium: 8,
+                large: 16,
+            },
+        };
+
+        const themeOverride = {
+            color: {
+                blue: "#0000ff",
+                red: "#ff0000",
+            },
+            spacing: {
+                small: 4,
+                medium: 8,
+            },
+        };
+
+        // Act
+        const result = mergeTheme(themeDefault, themeOverride);
+
+        // Assert
+        expect(result).toEqual({
+            color: {
+                blue: "#0000ff",
+                green: "#00ff00",
+                red: "#ff0000",
+            },
+            spacing: {
+                small: 4,
+                medium: 8,
+                large: 16,
+            },
+        });
+    });
+
+    it("should override the global tokens", () => {
+        // Arrange
+        const themeDefault = tokens;
+
+        // Only override the blue color
+        const themeOverride = {
+            color: {
+                blue: "#0000ff",
+            },
+        };
+
+        // Act
+        const result = mergeTheme(themeDefault, themeOverride);
+
+        // Assert
+        expect(result.color.blue).toEqual("#0000ff");
+    });
+});
```

Comment:
In `merge-theme.mdx`, you have a note saying

> Note that the first argument will define the
interface/contract that the second arg needs to follow.

What happens if the second argument doesn't follow the interface properly? Is there a test for that?

Reply:
TS will throw an error :)

e.g.
<img width="692" alt="Screenshot 2023-08-14 at 11 07 37 AM" src="https://github.com/Khan/wonder-blocks/assets/843075/1ff1ec35-b905-4366-a594-44d1233a8f23">

Reply:
Oh perfect!